scylla-cql/src/frame/response/authenticate.rs

```diff
@@ -42,3 +44,96 @@ impl AuthChallenge {
         })
     }
 }
+
```

Comment:
The `frame::response::authenticate` module is supposed to be related to the `AUTHENTICATE` frame in the CQL protocol. It's not a good place for the new authentication-related types that you introduced.

Let's add an `authentication` module in the root of the `scylla` crate and put those types there.